examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -26,18 +26,27 @@
 # %%
 # Load Ames Housing dataset
 # -------------------------
-# First, we load the ames housing data as a pandas dataframe. The features
+# First, we load the Ames Housing data as a pandas dataframe. The features
 # are either categorical or numerical:
 from sklearn.datasets import fetch_openml
 
 X, y = fetch_openml(data_id=41211, as_frame=True, return_X_y=True)
 
-n_categorical_features = (X.dtypes == "category").sum()
-n_numerical_features = (X.dtypes == "float").sum()
+# Take a subset of features of X to make the example faster to run
+categorical_features = X.select_dtypes(include="category").columns
+numerical_features = X.select_dtypes(include="number").columns
+
+n_columns = 10
+columns_subset = list(categorical_features[:n_columns]) + list(
+    numerical_features[:n_columns]
+)
+
+X = X[columns_subset]
+
 print(f"Number of samples: {X.shape[0]}")
 print(f"Number of features: {X.shape[1]}")
-print(f"Number of categorical features: {n_categorical_features}")
-print(f"Number of numerical features: {n_numerical_features}")
+print(f"Number of categorical features: {n_columns}")
+print(f"Number of numerical features: {n_columns}")
```

Comment:
This will become wrong if `n_columns` is adjusted to another value. Please move back `n_categorical_features` and `n_numerical_features` here instead.